### ISSUE TITLE
Windows build fails to embed the manifest

### ISSUE THREAD

**Issue by theantigod** [curl/curl#1832]
### I did this

I am currently building curl-7.55.1 for Windows 64bit.
This issue may have been detected but could not find any mention of it.  

mt.exe : general error c10100b1: Failed to load file "..\builds\libcurl-vc9-x64-debug-dll\curl.exe". The system cannot find the file specified.
NMAKE : fatal error U1077: '"C:\Program Files\Microsoft SDKs\Windows\v6.0A\bin\mt.EXE"' : return code '0x1f' Stop.
NMAKE : fatal error U1077: '"c:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\nmake.exe"' : return code '0x2'

In MakefileBuild.vc:

!IFDEF EMBED_MANIFEST
MANIFESTTOOL = mt -manifest $(DIRDIST)\\$(PROGRAM_NAME).manifest -outputresource:$(DIRDIST)\\$(PROGRAM_NAME);1
!ENDIF

Needs to be changed:  \bin is missing from the 2 paths.

MANIFESTTOOL = mt -manifest $(DIRDIST)\bin\\$(PROGRAM_NAME).manifest -outputresource:$(DIRDIST)\bin\\$(PROGRAM_NAME);1

**Comment by jay** [comment]
Thanks, fixed. I was able to reproduce using these commands in a Visual Studio 2008 x64 command prompt:

~~~
cd /d curldir
buildconf.bat
cd winbuild
nmake /f Makefile.vc mode=dll VC=9 ENABLE_IDN=no EMBED_MANIFEST=defined
~~~

Where did you find out about EMBED_MANIFEST? It is not documented AFAICT and I don't think anyone's using it in their build scripts or we would have caught this a long time ago.

**Comment by theantigod** [comment]
Yes, I understand how this would go unnoticed. I am a programmer and use make files instead of building with the Visual Studio IDE. I found that I had issues with curl.exe because I am not in the habit of copying the manifest with with the exe's. I have the mt command included in my own make files. So, I just looked through curl's makefile with the intention of adding the mt command and found that all that was needed was a define - EMBED_MANIFEST.My coworker is the one who usually builds curl for our use, and he has been manually embedding ‎the manifest with mt.exe after the build. But I needed a 64bit debug version for my R&D and I found this extra step annoying.Curl is a wonderful product. Thanks for taking the time to fix this.                                                                                                                                                                                                                                                                                                                                        Chris                                                                                                                                                                                                                From: Jay SatiroSent: Saturday, August 26, 2017 02:37To: curl/curlReply To: curl/curlCc: theantigod; AuthorSubject: Re: [curl/curl] Windows build fails to embed the manifest (#1832)Thanks, fixed. I was able to reproduce using these commands in a Visual Studio 2008 x64 command prompt:
cd /d curldir
buildconf.bat
cd winbuild
nmake /f Makefile.vc mode=dll VC=9 ENABLE_IDN=no EMBED_MANIFEST=defined

Where did you find out about EMBED_MANIFEST? It is not documented AFAICT and I don't think anyone's using it in their build scripts or we would have caught this a long time ago.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/curl/curl","title":"curl/curl","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/curl/curl"}},"updates":{"snippets":[{"icon":"PERSON","message":"@jay in #1832: Thanks, fixed. I was able to reproduce using these commands in a Visual Studio 2008 x64 command prompt:\r\n\r\n~~~\r\ncd /d curldir\r\nbuildconf.bat\r\ncd winbuild\r\nnmake /f Makefile.vc mode=dll VC=9 ENABLE_IDN=no EMBED_MANIFEST=defined\r\n~~~\r\n\r\nWhere did you find out about EMBED_MANIFEST? It is not documented AFAICT and I don't think anyone's using it in their build scripts or we would have caught this a long time ago."}],"action":{"name":"View Issue","url":"https://github.com/curl/curl/issues/1832#issuecomment-325094098"}}}
